File: lib/Conversation/Message.dart
class Message {
  final String id;
  final String uid;
  final String username;
  final DateTime timestamp;
  final String selectedImagePath;
  final String message;

  Message({
    required this.id,
    required this.uid,
    required this.username,
    required this.timestamp,
    required this.selectedImagePath,
    required this.message,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
        id: json['_id'] ?? '',
        uid: json['uid'] ?? '',
        username: json['username'] ?? '',
        timestamp: DateTime.parse(json['timestamp'] ?? ''),
        selectedImagePath: json['selectedImagePath'] ?? '',
        message: json['message'] ?? '',
    );
  }
}

File: server/Message.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const messageSchema = new Schema({
  id: String,
  uid: String,
  username: String,
  timestamp: { type: Date, default: Date.now },
  selectedImagePath: String,
  message: String,
});

const Message = mongoose.model('Message', messageSchema);

module.exports = Message;
